lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -141,6 +143,34 @@ static opt<std::string> AAWriterPath("aa-writer",
                                           "with annotated alias info."),
                                      value_desc("filename"));
 
+enum ABIEnforcementOption {
+  NoABIEnforcement = 0,
+  SoftABIEnforcement,
+  FullABIEnforcement,
+};
+
+using ABIOpt = ABIEnforcementOption;
+static opt<ABIOpt> ABIEnforcement("abi-enforcement-level",
+                                  desc("ABI refinement preferences."),
+                                  values(clEnumValN(NoABIEnforcement,
+                                                    "no",
+                                                    "Do not enforce "
+                                                    "ABI-specific "
+                                                    "information."),
+                                         clEnumValN(SoftABIEnforcement,
+                                                    "soft",
+                                                    "Enforce ABI-specific "
+                                                    "information, but allows "
+                                                    "for incompatibility, if "
+                                                    "unsure about the ABI."),
+                                         clEnumValN(FullABIEnforcement,
+                                                    "full",
+                                                    "Enforce ABI-specific "
+                                                    "information, and do not "
+                                                    "allow for any possible "
+                                                    "violations of the ABI "
+                                                    "found.")));
+
```

Comment:
After some testing, I feel like we may wish to default-initialize `ABIEnforcement` option to `FullABIEnforcement`, and fully leverage the ABI-specific layer. @aleclearmind, what do you think?